### DISCUSSION TITLE
Table footer with sum/average/custom

### DISCUSSION THREAD

**Discussion by AmosSzust** [quasarframework/quasar#15170] [General - Components / Directives / etc] [answered]
Hi

I'm using qTable and I was hoping there will be a footer with total/average/custom options per column but I read that I should use the before/after slots.

I prefer to use template v-slot:bottom so it will remain intact when scrolling up/down, but
1. If the column is hidden by the user, the relevant footer cell should be hidden as well.
2. It means I should total/average manually each column separately and manually and "attach" the result to each cell manually. Also, what if I have more than one page?
3. I need the total of specific columns and average of other columns so eventually it becomes cumbersome.
4. Maybe there are more issues that I will encounter only when (if?) I'll try to do this.

Are there plans to add such footer feature or maybe other people already did something similar and can share their experience of adding this footer behavior?

Thanks

**Comment by metalsadman** [comment, marked as answer]
you can't use `bottom` slot because it's a div not part of the html table, and it doesn't expose necessary scoped properties to rebuild a the columns (see slot api cards for the scope properties exposed by each slot), but you can still make use of the `bottom-row` slot and add css to position it sticky, combining docs examples based on your description see https://codepen.io/metalsadman/pen/ZEjbwwb.

**Reply by AmosSzust** [reply]
Thank you very much, the codepen example was very helpful.
